src/KeysManager.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\pubkey_encrypt\KeysManager.
+ */
+
+namespace Drupal\pubkey_encrypt;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\user\PrivateTempStoreFactory;
+use Drupal\user\UserInterface;
+
+/**
+ * Handles users' Public/Private key pairs.
+ */
+class KeysManager {
+  protected $entityTypeManager;
+  protected $tempStore;
+
+  /*
+   * Constructor with dependencies injected to it.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, PrivateTempStoreFactory $tempStoreFactory) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->tempStore = $tempStoreFactory->get('pubkey_encrypt');
+  }
+
+  /*
+   * Initialize all users' keys.
+   */
+  public function initializeAllUserKeys() {
+    $users = $this->entityTypeManager->getStorage('user')->loadMultiple();
+
+    foreach ($users as $user) {
+      $this->initializeUserKeys($user);
+    }
+  }
+
+  /*
+   * Initialize a specific user's keys.
+   */
+  public function initializeUserKeys(UserInterface $user) {
+    // Generate a Public/Private key pair.
+    $config = array(
+      "config" => "C:/xampp/apache/bin/openssl.cnf",
+      "digest_alg" => "sha512",
+      "private_key_bits" => 4096,
+      "private_key_type" => OPENSSL_KEYTYPE_RSA,
+    );
+    // Create the private and public key.
+    $res = openssl_pkey_new($config);
+    // Extract the private key.
+    openssl_pkey_export($res, $privKey, NULL, $config);
+    // Extract the public key.
+    $pubKey = openssl_pkey_get_details($res);
+    $pubKey = $pubKey["key"];
+
+    $privateKey = $privKey;
+    $publicKey = $pubKey;
+
+    // Set Public/Private keys.
+    $user
+      ->set('field_public_key', $publicKey)
+      ->set('field_private_key', $privateKey)
+      ->set('field_private_key_protected', 0)
+      ->save();
+  }
+
+  /*
+   * Protect a user keys with his credentials.
+   */
+  public function protectUserKeys(UserInterface $user, $credentials) {
+    $isProtected = $user->get('field_private_key_protected')
+      ->get(0)
+      ->getValue();
+    $isProtected = $isProtected['value'];
+    // Ensure that the keys have not already been protected.
+    if (!$isProtected) {
+      // Get original private key.
+      $privateKey = $user->get('field_private_key')
+        ->get(0)
+        ->getValue();
+      $privateKey = $privateKey['value'];
```

Comment:
Done. For reference:

'$isProtected = $user->get('field_private_key')->getString();'